src/v2/guide/composition.md

```diff
@@ -0,0 +1,529 @@
+
+``` js
+Vue.mixin({
+  created: function () {
+    console.log('A component was just created!')
+  }
+})
+```
+
+One appropriate use case, however, is to inject processing logic for custom options, such as in the [VueFire plugin](https://github.com/vuejs/vuefire#usage):
+
+``` js
+// Inject handler for a `firebase` custom option
+Vue.mixin({
+  created: function () {
+    var firebaseBindings = this.$options.firebase
+    if (firebaseBindings) {
+      // Add instance properties and listeners
+      // for data from firebase.
+    }
+  }
+})
+```
+
+Check out [VueFire's source](https://github.com/search?utf8=%E2%9C%93&q=%22Vue.mixin%22+repo%3Avuejs%2Fvuefire+path%3A%2Fsrc&type=Code) for a complete example.
+
+
+
+## Custom Merge Strategies
+
+Plugins often define custom options, which use the default strategy of overwriting any existing value. If you're the author a plugin where this is undesirable, you can define a **custom merge strategy** for your options with `Vue.config.optionMergeStrategies`.
+
+For example, if your plugin uses a `foo` option:
+
+``` js
+Vue.config.optionMergeStrategies.foo = function (newValue, baseValue) {
+  // Return the correctly merged value
+}
+```
+
+For most object-based options, you can simply use the same strategy used by `methods`:
+
+``` js
+var strategies = Vue.config.optionMergeStrategies
+strategies.foo = strategies.methods
+```
+
+A more advanced example can be found on [Vuex](https://github.com/vuejs/vuex)'s 1.x merging strategy:
+
+``` js
+const merge = Vue.config.optionMergeStrategies.computed
+Vue.config.optionMergeStrategies.vuex = function (newValue, baseValue) {
+  if (!newValue) return baseValue
+  if (!baseValue) return newValue
+  return {
+    getters: merge(newValue.getters, baseValue.getters),
+    state: merge(newValue.state, baseValue.state),
+    actions: merge(newValue.actions, baseValue.actions)
+  }
+}
+```
+
+
+
+## Wrapper Components
+
+As you've seen, advanced composition with extends and mixins can quickly grow complicated. One simpler alternative is creating **wrapper components**, which provide an interface for another component typically used as its root element.
+
+For example, the `ErrorNote` component we [created earlier with extension](#An-Example-Use-Case) could be refactored to:
+
+``` js
+var ErrorNote = {
+  components: {
+    AppNote: AppNote
+  },
+  template: `
+    <AppNote type="error">
+      <span slot="bold">Error!</span>
+      <slot/>
+    </AppNote>
+  `
+}
+```
+
+Instead of inheriting and building on the interface of a base component, this strategy does **not** provide direct access to the wrapped component. So unlike the `ErrorNote` using extension, this one has its own props (currently none) and its own slots (currently just the default slot).
+
+This is useful when we'd really like to _remove_ options from a base interface. For example, in the case of `ErrorNote`, it'd be very strange for it to be used like this:
```

Comment:
I think we need to be careful here. With all the changes we now avoided using the word "interface" as reference to the components API/props. Can we just replace `interface` with `component`?